hotham/src/asset_importer/mod.rs

```diff
@@ -229,18 +281,131 @@ fn load_node(
         world.insert_one(this_entity, Root {}).unwrap();
     }
 
-    for child in node_data.children() {
+    // If this node has corresponding collider geometry, add it in.
+    if let Some(collider) = get_collider_for_node(
+        node,
+        import_context,
+        this_entity,
+        local_transform.position(),
+    ) {
+        world.insert_one(this_entity, collider).unwrap();
+    }
+
+    // Now walk through each of this node's children and load them in.
+    for child in node.children() {
         load_node(&child, import_context, world, false);
     }
 
     this_entity
 }
 
-fn add_parents(
+/// Searches through the glTF document to find a mesh that can be used by Hotham to represent a collider, then creates one.
+///
+/// There are two kinds of colliders we're looking for:
+///
+/// - Walls, which are stored separate node with the same root name as some entity, eg. `cube` and `cube.HOTHAM_COLLIDER_WALL`
+/// - Sensors, which are their own separate nodes, eg. `phantom.HOTHAM_COLLIDER_SENSOR`
+fn get_collider_for_node(
+    node: &gltf::Node,
+    import_context: &mut ImportContext,
+    this_entity: Entity,
+    position: Isometry3<f32>,
+) -> Option<Collider> {
+    // First, get the name of the node, if it has one.
+    let node_name = node.name()?;
+
+    let (collider_node_name, mesh) = if node_name.contains(SENSOR_COLLIDER_TAG) {
+        (node_name, node.mesh()?)
+    } else {
+        find_wall_collider_for_node(node_name, import_context)?
+    };
+
+    // Build a collider using the mesh.
+    let shape = get_shape_from_mesh(mesh, import_context);
+    let collider_builder = rapier3d::geometry::ColliderBuilder::new(shape)
+        .user_data(this_entity.to_bits().get() as _)
+        .position(position);
+
+    let collider = if collider_node_name.contains(WALL_COLLIDER_TAG) {
+        println!(
+            "[HOTHAM_ASSET_IMPORTER] Created wall collider for model {}",
+            collider_node_name
+        );
+        collider_builder.sensor(false).build()
+    } else if collider_node_name.contains(SENSOR_COLLIDER_TAG) {
+        println!(
+            "[HOTHAM_ASSET_IMPORTER] Created sensor collider for model {}",
+            collider_node_name
+        );
+        collider_builder.sensor(true).build()
+    } else {
+        return None;
+    };
+
+    let handle = import_context.physics_context.colliders.insert(collider);
+
+    Some(Collider::new(handle))
+}
+
+fn find_wall_collider_for_node<'a>(
+    name: &'a str,
+    import_context: &'a ImportContext,
+) -> Option<(&'a str, gltf::Mesh<'a>)> {
+    // Create a pattern to search for the collider's name:
+    let wall_pattern = format!("{}.HOTHAM_COLLIDER_WALL", name);
+
+    // Iterate through each node to try and find the matching node, then fetch its mesh.
+    import_context.document.nodes().find_map(|n| {
+        if !n.name()?.contains(&wall_pattern) {
+            return None;
+        }
+
+        n.mesh().map(|mesh| (n.name().unwrap(), mesh))
```

Comment:
```suggestion
        if n.name()?.contains(&wall_pattern) {
            n.mesh().map(|mesh| (n.name().unwrap(), mesh))
        } else {
            None
        }
```

Reply:
I much prefer early returns to branches - way more readable

Reply:
You could still swap the order of them to get rid of the double negation.

```suggestion
        if n.name()?.contains(&wall_pattern) {
            return n.mesh().map(|mesh| (n.name().unwrap(), mesh));
        }
        None
```

Reply:
I think it's just not as clear that way - I like writing the "happy path" as the last statement of a branch